src/main/java/net/countercraft/movecraft/combat/features/directors/CannonDirectors.java

```diff
@@ -124,12 +125,16 @@ private void processTNT(@NotNull TNTPrimed tnt) {
         double horizontalSpeed = tntVector.length();
         tntVector = tntVector.normalize(); // you normalize it for comparison with the new direction to see if we are trying to steer too far
 
-        Block targetBlock = DirectorUtils.getDirectorBlock(p, CannonDirectorRange);
-        Vector targetVector;
-        if (targetBlock == null || targetBlock.getType().equals(Material.AIR)) // the player is looking at nothing, shoot in that general direction
-            targetVector = p.getLocation().getDirection();
-        else // shoot directly at the block the player is looking at (IE: with convergence)
-            targetVector = targetBlock.getLocation().toVector().subtract(tnt.getLocation().toVector());
+        // the player is looking at nothing, shoot in that general direction
+        Vector targetVector = p.getLocation().getDirection();
+
+        if (CannonDirectorRange >= 0) {
+            Block targetBlock = DirectorUtils.getDirectorBlock(p, CannonDirectorRange);
+            if (targetBlock != null && targetBlock.getType().equals(Material.AIR)) {
+                // shoot directly at the block the player is looking at (IE: with convergence)
+                targetVector = targetBlock.getLocation().toVector().subtract(tnt.getLocation().toVector());
+            }
+        }
```

Comment:
And here

Reply:
Because DirectorUtils will ALWAYS execute a raycast, regardless of what you give it, sure the relevant code here could be included there however.